Feature ut 30 fix calendar bugs

yogg-soggot opened the pull request.

yogg-soggot committed:
Fix empty task title possibility

yogg-soggot committed:
Fix current day events show

Current day events were not shown before you click on a day

yogg-soggot committed:
Fix not underlined event time

yogg-soggot committed:
Fix notification and repeating dialog style

yogg-soggot committed:
Fix wrong color on current day in next month

yogg-soggot committed:
Merge branch 'dev' into feature-UT_30-fix-calendar-bugs

# Conflicts:
#	.idea/gradle.xml
#	build.gradle
#	gradle/wrapper/gradle-wrapper.properties

VitalyPeryatin commented on app/src/main/java/com/tla/uchattut/presentation/schedule/ScheduleFragment.kt:
```diff
@@ -322,6 +329,14 @@ class ScheduleFragment : BaseFragment(), EventsRecyclerAdapter.OnEventItemClickL
         timePickerDialog.show()
     }
 
+    private fun onNotificationSelected(notifyBefore: String){
```
Проверь код через KtLint и отформатируй его
KtLint уже присоединен к проекту

VitalyPeryatin commented on app/src/main/java/com/tla/uchattut/presentation/schedule/ScheduleFragment.kt:
```diff
@@ -340,30 +355,39 @@ class ScheduleFragment : BaseFragment(), EventsRecyclerAdapter.OnEventItemClickL
 
     private fun addNewEvent() {
-        val newEvent = buildNewEvent()
+        val newEvent = try {
```
Очень не читаемая конструкция
Лучше то, что должно выполнится занести в блок try

VitalyPeryatin commented on app/src/main/java/com/tla/uchattut/presentation/schedule/ScheduleFragment.kt:
```diff
@@ -340,30 +355,39 @@ class ScheduleFragment : BaseFragment(), EventsRecyclerAdapter.OnEventItemClickL
-    )
+    private fun buildNewEvent(): EventPresentationModel {
+        val title = titleEditText.text.toString()
+        if(title == "") throw IllegalArgumentException("Title is empty")
```
Лучше сделать проверку через title.isBlank() 
Таким образом строки в которых например одни пробелы тоже не будут добавляться

VitalyPeryatin commented on app/src/main/java/com/tla/uchattut/presentation/schedule/dialogs/NotificationSelectorDialog.kt:
```diff
@@ -23,9 +26,23 @@ class NotificationSelectorDialog private constructor() : DialogFragment() {
+        super.onViewCreated(view, savedInstanceState)
+        rGroupNotification.setOnCheckedChangeListener{_, checkedId ->
+           when(checkedId) {
+               R.id.radioBtn5min -> onNotificationSelected("За 5 минут")
```
Лучше вынести эти строки в ресурсы

VitalyPeryatin commented on app/src/main/java/com/tla/uchattut/presentation/schedule/dialogs/RepeatingSelectorDialog.kt:
```diff
@@ -23,9 +26,22 @@ class RepeatingSelectorDialog private constructor() : DialogFragment() {
+        super.onViewCreated(view, savedInstanceState)
+        rGroupRepeating.setOnCheckedChangeListener{ _, checkedId ->
+            when(checkedId){
+                R.id.radioBtnNotRepeating -> onRepeatSelected("Не повторяется")
```
Строки в ресурсы

VitalyPeryatin commented on app/src/main/java/com/tla/uchattut/presentation/schedule/dialogs/RepeatingSelectorDialog.kt:
```diff
@@ -7,8 +7,11 @@ import android.view.ViewGroup
 
-class RepeatingSelectorDialog private constructor() : DialogFragment() {
+class RepeatingSelectorDialog private constructor(
+    private val onRepeatSelected: (repeat: String) -> Unit
```
Передавать что-либо в конструктор фрагментов нельзя
При пересоздании экрана всё будет падать

yogg-soggot committed:
Fix loading current day events after adding new to another day

- After adding new event, current day events was shown instead of the day on which events were added
- Code checked with KtLint

yogg-soggot committed:
Fix code according to code review
